make restclient config configurable

shiyan2016 opened the pull request:
<!--  Thanks for sending a pull request!  Here are some tips for you:
1. If this is your first time, read our contributor guidelines https://git.k8s.io/community/contributors/guide#your-first-contribution and developer guide https://git.k8s.io/community/contributors/devel/development.md#development-guide
2. If you want *faster* PR reviews, read how: https://git.k8s.io/community/contributors/guide/pull-requests.md#best-practices-for-faster-reviews
3. Add an entry to CHANGELOG.md if the PR represents a user-visible change.
4. If the PR is unfinished, see how to mark it: https://git.k8s.io/community/contributors/guide/pull-requests.md#marking-unfinished-pull-requests
-->

**What this PR does / why we need it**:
make restclient config configurable

**Which issue(s) this PR fixes** *(optional, in `fixes #<issue number>(, fixes #<issue_number>, ...)` format, will close the issue(s) when PR gets merged)*:
Fixes #

**Special notes for your reviewer**:

shiyan2016 commented:
/retest

shiyan2016 commented:
@hectorj2f Please help me to retrigger the test,thanks.

hectorj2f commented on cmd/controller-manager/app/controller-manager.go:
```diff
@@ -98,6 +99,8 @@ member clusters and do the necessary reconciliation`,
 	flags.StringVar(&kubeFedConfig, "kubefed-config", "", "Path to a KubeFedConfig yaml file. Test only.")
 	flags.StringVar(&kubeconfig, "kubeconfig", "", "Path to a kubeconfig. Only required if out-of-cluster.")
 	flags.StringVar(&masterURL, "master", "", "The address of the Kubernetes API server. Overrides any value in kubeconfig. Only required if out-of-cluster.")
+	flags.IntVar(&restConfigQPS, "rest-config-qps", 5, "QPS of rest config.")
```
Just to confirm, are these the default values ?

shiyan2016 replied:
yes, they are default values.

CharlesQQ replied:
1

hectorj2f commented:
I re-trigger it but we are always failing the docker image pulling issue. So we need to fix that first :/.

hectorj2f commented:
This problem is related to this issue https://github.com/kubernetes-sigs/kubefed/issues/1359. We need to work on that.

shiyan2016 commented:
> This problem is related to this issue #1359. We need to work on that.

ok, thanks.

CharlesQQ commented:
> /retest

1

hectorj2f commented:
Please, rebase to get the latest changes.

shiyan2016 commented:
> Please, rebase to get the latest changes.

Done it. Thanks.

xunpan commented on cmd/controller-manager/app/controller-manager.go:
```diff
@@ -98,6 +99,8 @@ member clusters and do the necessary reconciliation`,
 	flags.StringVar(&kubeFedConfig, "kubefed-config", "", "Path to a KubeFedConfig yaml file. Test only.")
 	flags.StringVar(&kubeconfig, "kubeconfig", "", "Path to a kubeconfig. Only required if out-of-cluster.")
 	flags.StringVar(&masterURL, "master", "", "The address of the Kubernetes API server. Overrides any value in kubeconfig. Only required if out-of-cluster.")
+	flags.IntVar(&restConfigQPS, "rest-config-qps", 5, "QPS of rest config.")
```
- if QPS is a float, it is better to use `Float32Var` instead.
- I suggest to use more detailed message for the description. E.g.:
  `QPS indicates the maximum QPS to the api-server from this client`
  `Maximum burst for throttle`

shiyan2016 replied:
@xunpan Done. Please review it again. Thanks.

xunpan reviewed (commented):
please squash 2 commits into 1.

xunpan commented:
/retest

xunpan commented:
/approve

xunpan commented:
@hectorj2f 
I don't know how to approve testing for this PR. Please help on this.

The code is fine to me.

irfanurrehman commented:
Have trigerred a build.

shiyan2016 commented:
```bash
[Fail] Federated "configmaps" [It] should be created, read, updated and deleted successfully 
/go/src/sigs.k8s.io/kubefed/test/common/crudtester.go:477

[Fail] Federated "replicasets.apps" [It] should be created, read, updated and deleted successfully 
/go/src/sigs.k8s.io/kubefed/test/common/crudtester.go:519
```
It seems not releated to my pr. @hectorj2f @xunpan @irfanurrehman

irfanurrehman commented:
@jimmidyson can this be related to the recent update to github actions? I did rerun this 3 times and each time, one of the first or second executed test failed with timeout (we set `single-call-timeout` of 1m for e2e test). Curiously all the later tests passed.

@shiyan2016 can you please push a "separate temperory commit" on this PR to update the [timeout](https://github.com/kubernetes-sigs/kubefed/blob/master/scripts/pre-commit.sh#L34) to a higher value, say 2m. This is only to determine if its just that the tests are executing slower then expected in the new environment, or its something else. Thanks in advance.

/hold

shiyan2016 commented:
> single-call-timeout

@irfanurrehman  Done for single-call-timeout to 2m. Please retriggle test. Thanks.
.

irfanurrehman commented:
/hold cancel
/lgtm
